### DISCUSSION TITLE
Rerender component on mutation cache changes

### DISCUSSION THREAD

**Discussion by spacepolice10** [TanStack/query#4348] [Q&A] [answered]
Hey there,

Is it possible to update mutation cache inside queryClient and make these changes affect the component state?

The idea: 

- There's an app where you could upload files.
- Every file triggers its own useMutation.
- I take every of them via queryClient as an array and show up in the interface the result of file uploading (the response from servers acquired by mutations).
- There's a button Delete before every file (its representation in the interface) which removes one by one. 
- I created an immer method that updates mutation cache array by splicing one of file (its data) but it doesn't work.

Probably changes inside mutation cache never affect component's state. Is there way to see updated mutation cache immediately after I have changed it manually?

**Comment by TkDodo** [comment, marked as answer]
mutations are meant to update data only. If you want to display a list of uploaded files, it sounds like you'd want a query that you can then invalidate after a mutation.
